actionpack/lib/action_dispatch/middleware/cookies.rb

```diff
@@ -458,6 +460,15 @@ def make_set_cookie_header(header)
         def write_cookie?(cookie)
           request.ssl? || !cookie[:secure] || always_write_cookie
         end
+
+        def set_expires_attribute(options)
+          expires_at, expires_in = options[:expires_at], options[:expires_in]
```

Comment:
> only use :expires, but make both expires: DateTime and expires: Duration DWIM.

Ah yeah, let's just go with that here 👍

Then we only have to convert `expires: 1.hour` into an absolute time via `options[:expires] = options[:expires].from_now if options[:expires].respond_to?(:from_now)`.

And pass it in to the verifier/encryptor via `expires_at: options[:expires]`.